### PR TITLE
[release/5.0] Align CODEOWNERS with 'main'

### PR DESCRIPTION
**Pull request by dougbu**
- add and remove many assignments
- mostly match 'main' version less missing directories

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
+/.devcontainer/                                                                   @captainsafia
+/.github/                                                                         @dotnet/aspnet-build
+/.github/*_TEMPLATE/                                                              @dotnet/aspnet-build @mkArtakMSFT
+/.github/workflows/                                                               @dotnet/aspnet-build @tratcher
+/docs/                                                                            @captainsafia @mkArtakMSFT
+/eng/                                                                             @dotnet/aspnet-build
+/eng/common/                                                                      @dotnet-maestro-bot
+/eng/Versions.props                                                               @dotnet-maestro-bot @dougbu
+/eng/Version.Details.xml                                                          @dotnet-maestro-bot @dougbu
+/src/Components/                                                                  @dotnet/aspnet-blazor-eng @SteveSandersonMS
```
Comment (dougbu) [inline .github/CODEOWNERS:16]: @SteveSandersonMS you don't have this separate assignment in 'release/6.0', 'main' or #35950. Is this needed here❔

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+/src/Components/**/PublicAPI.*Shipped.txt                                         @dotnet/aspnet-api-review @dotnet/aspnet-blazor-eng
```
Comment (dougbu) [inline .github/CODEOWNERS:17]: Seems we totally missed including @dotnet/aspnet-api-review for these files in this branch☹️

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,27 +1,40 @@
 # Users referenced in this file will automatically be requested as reviewers for PRs that modify the given paths.
 # See https://help.github.com/articles/about-code-owners/
 
-/global.json                    @aspnet/build
```
Comment (dougbu) [inline .github/CODEOWNERS:4]: @Pilchie same question here as in #35950 about you owning the world

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
-/eng/Version.Details.xml        @dotnet-maestro-bot @dougbu
-/src/Components/                @SteveSandersonMS @dotnet/aspnet-blazor-eng
-/src/DefaultBuilder/            @tratcher
-/src/Hosting/                   @tratcher
-/src/Http/                      @tratcher @jkotalik
-/src/Http/Routing/              @javiercn
-/src/Middleware/                @tratcher
-/src/Middleware/HttpsPolicy/    @jkotalik
-/src/Middleware/Rewrite/        @jkotalik
-# /src/ProjectTemplates/        @ryanbrandenburg
```
Comment (dougbu) [inline .github/CODEOWNERS:20]: Same question (I guess to leads) about filling in src/* gaps as in #35950

File: .github/CODEOWNERS (hunk 3 above)
Comment (Pilchie) [reply .github/CODEOWNERS:4]: > @Pilchie same question here as in #35950 about you owning the world

Fine with me.

**Comment by dougbu** [comment]
Will rerun once #35637 is in

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**